### DISCUSSION TITLE
Setting an env var override in a pydantic-settings nested model

### DISCUSSION THREAD

**Discussion by michaeltinsley** [pydantic/pydantic#8989] [Question]
I'm trying to get an alias working for a nested model with Pydantic Settings.

For example:

```python
from pydantic_settings import BaseSettings, SettingsConfigDict
from pydantic import BaseModel, Field


class NestedModel(BaseModel):

    name: str = Field(alias="SOME_GLOBAL_VAR")

class Settings(BaseSettings):
    x: str = "some value"
    nested_model: NestedModel

    model_config = SettingsConfigDict(
        env_prefix="MY_APP_",
        env_nested_delimiter="__",
    )
```

I want to be able to set `export SOME_GLOBAL_VAR=some_value`, and then have the model parsed to 
```
Settings(x='some value', nested_model=NestedModel(name='some_value'))
```

But I can't get it to work at all. Basically the nested model should ignore the env_prefix and the delimiter and just pick up the aliased env var.

Any ideas if this is possible and if so how it can be achieved?

**Comment by hramezani** [comment]
It is a nested model and `pydantic-settings` collects env variables from the parent class.
As you set the `env_prefix="MY_APP_"` and `env_nested_delimiter="__"` you can still collect the field value by the ‍`alias` you need to change the alias to lowercase `some_global_var` and then you need to set the `MY_APP_NESTED_MODEL__SOME_GLOBAL_VAR` env variable.

**Comment by galak75** [comment]
I am facing the exact same issue. Overriding the environment variable name seems to be legitimate : it could be an environment variable coming from a pipeline secret, and as an application developer, I unfortunately can not choose the variable name.
Therefore, I would like to override its name (git it an alias) without changing my application settings structure... 

The same question has been asked here : https://github.com/pydantic/pydantic/discussions/8221

Could this question lead to a feature request ?

**Comment by galak75** [comment]
> 
> Any ideas if this is possible and if so how it can be achieved?

@michaeltinsley : I think you have to make your `NestedModel` extend `BaseSettings` to make it work as you expect
